lib/streamlit/external/langchain/streamlit_callback_handler.py

```diff
@@ -170,18 +160,21 @@ def last_tool(self) -> Optional[ToolRecord]:
         return self._last_tool
 
     def _reset_llm_token_stream(self) -> None:
+        if self._llm_token_stream_placeholder is not None:
+            self._llm_token_stream_placeholder.markdown(self._llm_token_stream)
+
         self._llm_token_stream = ""
-        self._llm_token_writer_idx = None
+        self._llm_token_stream_placeholder = None
 
     def on_llm_start(self, serialized: Dict[str, Any], prompts: List[str]) -> None:
         self._reset_llm_token_stream()
 
     def on_llm_new_token(self, token: str, **kwargs: Any) -> None:
         # This is only called when the LLM is initialized with `streaming=True`
         self._llm_token_stream += _convert_newlines(token)
-        self._llm_token_writer_idx = self._container.markdown(
-            self._llm_token_stream, index=self._llm_token_writer_idx
-        )
+        if self._llm_token_stream_placeholder is None:
+            self._llm_token_stream_placeholder = self._container.empty()
+        self._llm_token_stream_placeholder.markdown(self._llm_token_stream + "▕")
```

Comment:
What's the reason we are adding this `" |"` at the end?

Reply:
It's a typing indicator like you have with ChatGPT or, in general, in editors.